sotopia/benchmark/cli.py

```diff
@@ -0,0 +1,297 @@
+from datetime import datetime
+from datasets import load_dataset
+from sotopia.database.persistent_profile import EnvironmentList
+import asyncio
+import logging
+from typing import Generator, cast
+
+from logging import FileHandler
+from rich.logging import RichHandler
+
+from tqdm import tqdm
+
+from sotopia.agents import LLMAgent
+from sotopia.database import (
+    AgentProfile,
+    EnvAgentComboStorage,
+    EnvironmentProfile,
+    EpisodeLog,
+)
+from sotopia.database.serialization import get_rewards_from_episode
+from sotopia.envs.evaluators import (
+    ReachGoalLLMEvaluator,
+    RuleBasedTerminatedEvaluator,
+)
+from sotopia.envs.parallel import ParallelSotopiaEnv
+from sotopia.generation_utils.generate import LLM_Name
+from sotopia.messages import AgentAction, Observation
+from sotopia.samplers import (
+    BaseSampler,
+    EnvAgentCombo,
+)
+from sotopia.server import run_async_server
+
+import typer
+from pathlib import Path
+
+app = typer.Typer()
+
+
+def check_existing_episodes(
+    env_id: str,
+    agent_ids: list[str],
+    models: dict[str, LLM_Name],
+    tag: str | None = None,
+) -> bool:
+    if tag:
+        existing_episode = EpisodeLog.find(
+            (EpisodeLog.environment == env_id) & (EpisodeLog.tag == tag)
+        ).all()
+    else:
+        existing_episode = EpisodeLog.find(EpisodeLog.environment == env_id).all()
+    if existing_episode:
+        for episode in existing_episode:
+            assert isinstance(episode, EpisodeLog), "episode should be an EpisodeLog"
+            if episode.agents == agent_ids and episode.models == list(models.values()):
```

Comment:
Change this line so that checking existing episodes doesn't reply on the ordering of the dict. You can do episode.models == list(models[key] for key in sorted(models.keys()))

Reply:
Simply doing that would not work in our case, I somehow need to rewrite the `_iterate_all_env_agent_combo_not_in_db` function, which should take more time on my side.